File: project_euler/Problem001/Problem001.js
function Problem001() {
    let target = 1000
    let result = getSummation(target, 3) + getSummation(target, 5) - getSummation(target, 3 * 5)
    console.log(result)
}

function getSummation(border, factor) {
    let count = Math.floor((border - 1) / factor)
    return factor * count * (count + 1) >>> 1
}

module.exports = {
    Problem001 : Problem001
};

File: project_euler/Problem001/Problem001.c
#include <stdio.h>

static int get_summation(int border, int factor);

void Problem001(void) {
    int target = 1000;
    int result = get_summation(target, 3) + get_summation(target, 5) - get_summation(target, 3 * 5);
    printf("%d\n", result);
}

static int get_summation(int border, int factor) {
    int count = (border - 1) / factor;
    return factor * count * (count + 1) >> 1;
}
